kani-compiler/src/codegen_cprover_gotoc/utils/utils.rs

```diff
@@ -46,18 +46,20 @@ impl<'tcx> GotocCtx<'tcx> {
     /// This means that if the unimplemented feature is dynamically used by the code being verified, we will see an assertion failure.
     /// If it is not used, we the assertion will pass.
     /// This allows us to continue to make progress parsing rust code, while remaining sound (thanks to the `assert(false)`)
-    ///
-    /// TODO: https://github.com/model-checking/kani/issues/8 assume the required validity constraints for the nondet return
-    /// TODO: https://github.com/model-checking/kani/issues/9 Have a parameter that decides whether to `assume(0)` to block further traces or not
     pub fn codegen_unimplemented(
         &mut self,
         operation_name: &str,
         t: Type,
         loc: Location,
         url: &str,
     ) -> Expr {
-        // We should possibly upgrade this to a warning in the future, but for now emit at least something
+        // Save this occurrence so we can emit a warning in the compilation report.
         debug!("codegen_unimplemented: {} at {}", operation_name, loc.short_string());
+        let key: InternedString = operation_name.into();
+        if !self.unsupported_constructs.contains_key(&key) {
+            self.unsupported_constructs.insert(key.clone(), Vec::new());
```

Comment:
I think the `clone` is not needed, because `InternedString` is `Copy`

Reply:
k